src/assets/js/app.js

```
@@ -419,6 +420,22 @@ export default function() {
         render();
     }
 
+    function handleHTTPErrors(error) {
```

@djmiley it actually handles both.
Previously error handling was just 
```JavaScript
statusText = error.statusText || 'Unknown reason.';
```
This was fine for the normal error codes, but the CloudFlare specific errors are not returned in the same way.
So to handle this problem, I added this function which checks for the specific CloudFlare error codes (520 - 526) and handles them appropriately.
If an error code falls outside this range, the error is just handled the same way as the previous implemented